### DISCUSSION TITLE
Contributing in Python

### DISCUSSION THREAD

**Discussion by nemhods** [windmill-labs/windmill#1752] [Questions]
Hey, 
I couldn't help but notice, all things on the hub are in typescript (as far as I could see at least).

I have a few things in mind that I might want to contribute (Rocket.Chat, Elasticsearch, maybe even LDAP?) but it would all be using python. Is python welcome as a contribution language, and will it have a future? It says Deno is recommended, but I'd say I'd rather there be pre-made scripts in some language than none at all.

Also, how does one go about to add a whole new integration: create the resource type first, wait for it to be approved, then post the scripts to the hub? How would the dev team even judge the usefulness and completeness of a resource type without the accompanying scripts?
Thanks!

**Comment by rubenfiszel** [comment]
Hi @nemhods ,

First thanks a lot for willing to contribute, it is very appreciated.

The reason why deno is prefered is simply because of performance reason. A simple deno script can take 50ms end-to-end versus python which will take more around 250ms. Does it really matter in the end ? Not so much and we would of course prefer 1 integration in python than 0 in deno. Also, once 1 integration exist in python, porting it in typescript if it's highly used and a performance bottleneck not an issue.

So yes, we would love your contribution even in python.
